src/lib/resolve-ipld-path.js

```diff
@@ -64,37 +67,45 @@ export default async function resolveIpldPath (ipld, sourceCid, path, nodes = []
   if (link) {
     pathBoundaries.push(link)
     // Go again, using the link.target as the sourceCid, and the remainderPath as the path.
-    return resolveIpldPath(ipld, link.target, remainderPath, nodes, pathBoundaries)
+    return resolveIpldPath(ipfs, link.target, remainderPath, nodes, pathBoundaries)
   }
   // we made it to the containing node. Hand back the info
   const canonicalPath = path ? `${sourceCidStr}${path}` : sourceCidStr
   const targetNode = node
+
   return { targetNode, canonicalPath, localPath: path, nodes, pathBoundaries }
 }
 
 /**
  * @function ipldGetNodeAndRemainder
- * @param {IpldInterface} ipld
+ * @param {typeof import('@helia/interface').Helia} helia
  * @param {string} sourceCid
  * @param {string} path
  * @returns
  */
-export async function ipldGetNodeAndRemainder (ipld, sourceCid, path) {
+export async function ipldGetNodeAndRemainder (helia, sourceCid, path) {
   if (sourceCid == null) {
     throw new Error('sourceCid is null')
   }
-  // TODO: find out why ipfs.dag.get with localResolve never resolves.
-  // const {value, remainderPath} = await getIpfs().dag.get(sourceCid, path, {localResolve: true})
+  let cidInstance = CID.asCID(sourceCid)
+  if (cidInstance === null) {
+    cidInstance = CID.parse(sourceCid)
+  }
+  const codecWrapper = await getCodecForCid(cidInstance)
+  const encodedValue = await getRawBlock(helia, cidInstance)
+  const value = codecWrapper.decode(encodedValue)
 
-  // TODO: use ipfs.dag.get when it gets ipld super powers
-  // SEE: https://github.com/ipfs/js-ipfs-api/pull/755
-  // const {value} = await getIpfs().dag.get(sourceCid)
+  const codecWrapperResolveResult = await codecWrapper.resolve(path || '/', encodedValue)
+  const { remainderPath, resolve: resolveValue } = codecWrapperResolveResult
+
+  if (resolveValue?.Hash != null) {
+    // This is a PBLink, and we should resolve that link so we're returning PBNodes not PBLinks
+    return await ipldGetNodeAndRemainder(helia, value.Hash, remainderPath)
+  }
```

Comment:
The recursion of ipld pathing happens at resolveIpldPath, which means we can't resolve dag-pb pathing inside the resolve function directly.. so until a refactor, this is the fix that I spent more time than I care to admit on.